### ISSUE TITLE
Do not return ID on Record.update() when updatablePrimaryKeys is set to false

### ISSUE THREAD

**Issue by fedotxxl** [jOOQ/jOOQ#5940]
Hello. We use SQLServer. JOOQ generates additional uselsess `SELECT` query on each `record.update` even when ID was not generated. Is it possible to disable it for all / specific `record.update` calls?

**Comment by lukaseder** [comment]
Thanks for reporting. This indeed doesn't make any sense for `UPDATE` statements. The `RETURNING` clause was added with #1859 as some users request fetching back all trigger-generated values after any type of `UpdatableRecord.store` operation. But if the setting from #1859 is turned off, then fetching the ID doesn't make sense, as an identity value shouldn't be changed.

We'll look into this.

As a workaround, you can create an actual `UPDATE` statement:

```java
ctx.update(TABLE)
   .set(record)
   .where(TABLE.ID.eq(record.get(TABLE.ID)))
   .execute();
```

Or use this:

```java
ctx.executeUpdate(record);
```

**Comment by lukaseder** [comment]
This behaviour can be tied to the existing `Settings.updatablePrimaryKeys` value. If it is set to true, then the status quo seems reasonable as users who are trying to update the primary key / identity value may expect it to be re-generated by the database, whereas in your case, it is safe to assume that it won't change on update. The default for `Settings.updatablePrimaryKeys == false`

**Comment by lukaseder** [comment]
Hmm, I don't have a good feeling reusing this `updatablePrimaryKeys` flag for this purpose. I'd rather introduce new flags

**Comment by lukaseder** [comment]
This fix has now been released for jOOQ 3.9.5. Your further feedback would be very welcome.
